service/controller/v3/resource/vnetpeering/current.go

```diff
@@ -0,0 +1,87 @@
+package vnetpeering
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/controller/v3/key"
+)
+
+// GetCurrentState retrieve the current host cluster virtual network peering
+// resource from azure.
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return network.VirtualNetworkPeering{}, microerror.Mask(err)
+	}
+
+	// In order to make vnet peering work we need a virtual network which we can
+	// use to peer. In case there is no virtual network yet we cancel the resource
+	// and try again on the next resync period. This is a classical scenario on
+	// guest cluster creation. If we would not check for the virtual network
+	// existence the client calls of CreateOrUpdate would fail with not found
+	// errors.
+	if !key.IsDeleted(customObject) {
```

Comment:
I think these checks should be the exception and they are not really necessary here, right?